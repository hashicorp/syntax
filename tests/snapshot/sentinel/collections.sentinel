# https://developer.hashicorp.com/sentinel/docs/language/collection-operations

# filter
filter list as value { condition }      // Single-iterator, list
filter list as idx, value { condition } // Double-iterator, list

filter map as key { condition }         // Single-iterator, map
filter map as key, value { condition }  // Double-iterator, map

# map
l = [1, 2]
r = map l as v { v % 2 } // [false, true]

m = { "a": "foo", "b": "bar" }
r = map m as k, v { v } // ["foo", "bar"]
