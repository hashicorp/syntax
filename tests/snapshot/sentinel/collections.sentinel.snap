># https://developer.hashicorp.com/sentinel/docs/language/collection-operations
#^ source.sentinel comment.line.number-sign.sentinel punctuation.definition.comment.sentinel
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.sentinel comment.line.number-sign.sentinel
>
># filter
#^ source.sentinel comment.line.number-sign.sentinel punctuation.definition.comment.sentinel
# ^^^^^^^ source.sentinel comment.line.number-sign.sentinel
>filter list as value { condition }      // Single-iterator, list
#^^^^^^ source.sentinel keyword.control.sentinel
#      ^^^^^^ source.sentinel
#            ^^ source.sentinel keyword.other.sentinel
#              ^^^^^^^^^^^^^^^^^^^^^^^^^^ source.sentinel
#                                        ^^ source.sentinel comment.line.double-slash.sentinel punctuation.definition.comment.sentinel
#                                          ^^^^^^^^^^^^^^^^^^^^^^ source.sentinel comment.line.double-slash.sentinel
>filter list as idx, value { condition } // Double-iterator, list
#^^^^^^ source.sentinel keyword.control.sentinel
#      ^^^^^^ source.sentinel
#            ^^ source.sentinel keyword.other.sentinel
#              ^^^^^^^^^^^^^^^^^^^^^^^^^^ source.sentinel
#                                        ^^ source.sentinel comment.line.double-slash.sentinel punctuation.definition.comment.sentinel
#                                          ^^^^^^^^^^^^^^^^^^^^^^ source.sentinel comment.line.double-slash.sentinel
>
>filter map as key { condition }         // Single-iterator, map
#^^^^^^ source.sentinel keyword.control.sentinel
#      ^ source.sentinel
#       ^^^ source.sentinel keyword.control.sentinel
#          ^ source.sentinel
#           ^^ source.sentinel keyword.other.sentinel
#             ^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.sentinel
#                                        ^^ source.sentinel comment.line.double-slash.sentinel punctuation.definition.comment.sentinel
#                                          ^^^^^^^^^^^^^^^^^^^^^ source.sentinel comment.line.double-slash.sentinel
>filter map as key, value { condition }  // Double-iterator, map
#^^^^^^ source.sentinel keyword.control.sentinel
#      ^ source.sentinel
#       ^^^ source.sentinel keyword.control.sentinel
#          ^ source.sentinel
#           ^^ source.sentinel keyword.other.sentinel
#             ^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.sentinel
#                                        ^^ source.sentinel comment.line.double-slash.sentinel punctuation.definition.comment.sentinel
#                                          ^^^^^^^^^^^^^^^^^^^^^ source.sentinel comment.line.double-slash.sentinel
>
># map
#^ source.sentinel comment.line.number-sign.sentinel punctuation.definition.comment.sentinel
# ^^^^ source.sentinel comment.line.number-sign.sentinel
>l = [1, 2]
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^^ source.sentinel
#     ^ source.sentinel constant.numeric.number.sentinel
#      ^^ source.sentinel
#        ^ source.sentinel constant.numeric.number.sentinel
#         ^^ source.sentinel
>r = map l as v { v % 2 } // [false, true]
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^ source.sentinel
#    ^^^ source.sentinel keyword.control.sentinel
#       ^^^ source.sentinel
#          ^^ source.sentinel keyword.other.sentinel
#            ^^^^^^^ source.sentinel
#                   ^ source.sentinel keyword.operator.symbol.sentinel
#                    ^ source.sentinel
#                     ^ source.sentinel constant.numeric.number.sentinel
#                      ^^^ source.sentinel
#                         ^^ source.sentinel comment.line.double-slash.sentinel punctuation.definition.comment.sentinel
#                           ^^^^^^^^^^^^^^ source.sentinel comment.line.double-slash.sentinel
>
>m = { "a": "foo", "b": "bar" }
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^^^ source.sentinel
#      ^ source.sentinel string.quoted.double.untitled
#       ^ source.sentinel string.quoted.double.untitled
#        ^ source.sentinel string.quoted.double.untitled
#         ^^ source.sentinel
#           ^ source.sentinel string.quoted.double.untitled
#            ^^^ source.sentinel string.quoted.double.untitled
#               ^ source.sentinel string.quoted.double.untitled
#                ^^ source.sentinel
#                  ^ source.sentinel string.quoted.double.untitled
#                   ^ source.sentinel string.quoted.double.untitled
#                    ^ source.sentinel string.quoted.double.untitled
#                     ^^ source.sentinel
#                       ^ source.sentinel string.quoted.double.untitled
#                        ^^^ source.sentinel string.quoted.double.untitled
#                           ^ source.sentinel string.quoted.double.untitled
#                            ^^^ source.sentinel
>r = map m as k, v { v } // ["foo", "bar"]
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^ source.sentinel
#    ^^^ source.sentinel keyword.control.sentinel
#       ^^^ source.sentinel
#          ^^ source.sentinel keyword.other.sentinel
#            ^^^^^^^^^^^^ source.sentinel
#                        ^^ source.sentinel comment.line.double-slash.sentinel punctuation.definition.comment.sentinel
#                          ^^^^^^^^^^^^^^^ source.sentinel comment.line.double-slash.sentinel
>